pkg/sql/colexec/top/top.go

```diff
@@ -61,47 +62,46 @@ func Prepare(proc *process.Process, arg any) (err error) {
 	return nil
 }
 
-func Call(idx int, proc *process.Process, arg any, isFirst bool, isLast bool) (process.ExecStatus, error) {
-	ap := arg.(*Argument)
+func Call(ins *vm.Instruction, proc *process.Process) (*batch.Batch, error) {
+	child := ins.Children[0]
+	ap := ins.Arg.(*Argument)
 	ctr := ap.ctr
-	anal := proc.GetAnalyze(idx)
+	anal := proc.GetAnalyze(ins.Idx)
 	anal.Start()
 	defer anal.Stop()
+
+	// special: limit == 0, return empty batch
+	if ap.Limit == 0 {
+		return nil, nil
+	}
+
+	// operator processing loop
 	for {
 		switch ctr.state {
 		case Build:
-			bat := proc.InputBatch()
-			if bat == nil {
-				ctr.state = Eval
-				continue
-			}
-			if bat.IsEmpty() {
-				proc.PutBatch(bat)
-				proc.SetInputBatch(batch.EmptyBatch)
-				return process.ExecNext, nil
-			}
-			if ap.Limit == 0 {
-				proc.PutBatch(bat)
-				proc.SetInputBatch(nil)
-				return process.ExecStop, nil
+			bat, err := vm.InstructionCall(child, proc)
```

Comment:
There is really no need for this Build and Eval.

It is a simple 

for not done {
          b, e := vm.InstructionCall(child, proc);
          ctr.build(b)
}

then eval and return batch by batch.